components/places/android/src/main/java/org/mozilla/places/PlacesConnection.kt

```diff
@@ -393,20 +456,12 @@ data class SearchResult(
 ) {
     companion object {
         fun fromJSON(jsonObject: JSONObject): SearchResult {
-            fun stringOrNull(key: String): String? {
-                return try {
-                    jsonObject.getString(key)
-                } catch (e: JSONException) {
-                    null
-                }
-            }
-
             return SearchResult(
                 searchString = jsonObject.getString("search_string"),
                 url = jsonObject.getString("url"),
                 title = jsonObject.getString("title"),
                 frecency = jsonObject.getLong("frecency"),
-                iconUrl = stringOrNull("icon_url")
+                iconUrl = stringOrNull(jsonObject, "icon_url")
```

Comment:
Does `jsonObject.optString` not work?

Reply:
Nope! optString returns the empty string if it's not present. Annoying, right?